src/components/AddItem.jsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useState } from 'react';
+import {
+	Box,
+	Drawer,
+	DrawerBody,
+	DrawerHeader,
+	DrawerOverlay,
+	DrawerContent,
+	DrawerCloseButton,
+	FormControl,
+	FormLabel,
+	RadioGroup,
+	Radio,
+	Input,
+	Button,
+	useToast,
+	Flex,
+	Stack,
+	Heading,
+	Text,
+} from '@chakra-ui/react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({
+	data,
+	listId,
+	isOpen,
+	onClose,
+	btnRef,
+	initialValue,
+}) {
+	const [timeframe, setTimeframe] = useState('7');
+	const [itemName, setItemName] = useState(initialValue);
+	const toast = useToast();
+
+	const onItemChange = (e) => setItemName(e.target.value);
+
+	// function to validate itemName input that returns error messages if any - called onFormSubmit
+	function validateItemInput(data, trimmedItemName) {
+		// 1- checks for empty inputs
+
+		if (!trimmedItemName) {
+			toast({
+				title: 'No item name entered',
+				description: `Please enter an item name.`,
+				variant: 'errorToast',
+			});
+			return false;
+		}
+
+		// 2- checks for invalid characters
+
+		if (trimmedItemName.search(/[^a-z0-9'&-\s]/i) !== -1) {
+			toast({
+				title: 'Invalid item name entered',
+				description: `Please only use alphanumeric, ', -, or & characters in your item name.`,
+				variant: 'errorToast',
+			});
+			return false;
+		}
+
+		const normalizedItemName = trimmedItemName
+			.replace(/[^a-z0-9'&-]/gi, '')
+			.toLowerCase();
+
+		const potentialMatch = data.find(
+			(item) =>
+				item.name.replaceAll(' ', '').toLowerCase() === normalizedItemName,
+		);
+
+		// 3- checks for potential match including exact match and match after removing special characters and spaces
+
+		if (potentialMatch) {
+			toast({
+				title: 'Item already on list',
+				description: `You are already tracking ${potentialMatch.name}.`,
+				variant: 'errorToast',
+			});
+			return false;
+		}
+		return true;
```

Comment:
![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/99099243/f6a8ff19-144e-4600-ba81-ef4f86668201)

Miso is already on the list. I am expecting to see both `Please only use alphanumeric, ', -, or & characters in your item name.` and  `Miso** is already on your list as Miso.` Would probably be best to have as two separate toasts. In the original implementation we implemented as an array of error messages which was later looped. I think the same pattern can be followed but to return multiple toasts with the error messages.

For reference:
![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/99099243/2e43f5ff-6550-457c-ace2-48c7fdd26b93)